packages/jest-resolve/src/resolver.ts

```diff
@@ -485,16 +489,9 @@ export default class Resolver {
   }
 
   getMockModule(from: string, name: string): string | null {
-    const mock = this._moduleMap.getMockModule(name);
-    if (mock) {
-      return mock;
-    } else {
-      const moduleName = this.resolveStubModuleName(from, name);
```

Comment:
@mrazauskas I reverted the major refactors but also refactored the inner logic of `Resolver.getMockModule` to be more in line to the method's function. Mostly because keeping it as is forces the logic in `Runtime.requireMock` to be too convoluted.

It would affect users who use it for the highlighted branch to get data about the original unmocked module.